src/test/java/org/folio/circulation/support/fetching/PageableFetcherTest.java

```diff
@@ -0,0 +1,123 @@
+package org.folio.circulation.support.fetching;
+
+import static java.util.concurrent.CompletableFuture.completedFuture;
+import static java.util.stream.Collectors.toList;
+import static java.util.stream.IntStream.range;
+
+import static org.folio.circulation.support.http.client.CqlQuery.noQuery;
+import static org.folio.circulation.support.http.client.PageLimit.limit;
+import static org.folio.circulation.support.results.Result.failed;
+import static org.folio.circulation.support.results.Result.ofAsync;
+import static org.hamcrest.CoreMatchers.containsString;
+import static org.hamcrest.CoreMatchers.is;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.spy;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+import static org.mockito.internal.verification.VerificationModeFactory.noInteractions;
+import static org.mockito.internal.verification.VerificationModeFactory.times;
+
+import java.util.concurrent.CompletableFuture;
+
+import org.folio.circulation.domain.MultipleRecords;
+import org.folio.circulation.support.ServerErrorFailure;
+import org.folio.circulation.support.http.client.PageLimit;
+import org.folio.circulation.support.results.Result;
+import org.junit.Test;
+
+public class PageableFetcherTest {
+  @Test
+  public void canProcessPages() {
```

Comment:
I know I encouraged the use of `can`, I think your instincts for `should` were good. Do you think `can` fits better here?

```suggestion
  public void shouldProcessPages() {
```

Reply:
Renamed to should.